packages/app-store/alby/lib/PaymentService.ts

```diff
@@ -0,0 +1,133 @@
+import type z from "zod";
+
+import prisma from "@calcom/prisma";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { IAbstractPaymentService } from "@calcom/types/PaymentService";
+
+import { albyCredentialKeysSchema } from "./albyCredentialKeysSchema";
+
+export class PaymentService implements IAbstractPaymentService {
+  private credentials: z.infer<typeof albyCredentialKeysSchema> | null;
+
+  constructor(credentials: { key: Prisma.JsonValue }) {
+    const keyParsing = albyCredentialKeysSchema.safeParse(credentials.key);
+    if (keyParsing.success) {
+      this.credentials = keyParsing.data;
+    } else {
+      this.credentials = null;
+    }
+  }
+
+  async create(
+    payment: Pick<Prisma.PaymentUncheckedCreateInput, "amount" | "currency">,
+    bookingId: Booking["id"]
+  ) {
+    try {
+      const booking = await prisma.booking.findFirst({
+        select: {
+          uid: true,
+          title: true,
+        },
+        where: {
+          id: bookingId,
+        },
+      });
+      if (!booking || !this.credentials?.account_lightning_address) {
+        throw new Error();
+      }
+
+      const uid = uuidv4();
+
+      const lightningAddress = new LightningAddress(this.credentials.account_lightning_address);
+      await lightningAddress.fetch();
+      const invoice = await lightningAddress.requestInvoice({
+        satoshi: payment.amount,
+        payerdata: {
+          appId: "cal.com",
+          referenceId: uid,
+        },
+      });
+      console.log("Created invoice", invoice, uid);
+
+      const paymentData = await prisma.payment.create({
+        data: {
+          uid,
+          app: {
+            connect: {
+              slug: "alby",
+            },
+          },
+          booking: {
+            connect: {
+              id: bookingId,
+            },
+          },
+          amount: payment.amount,
+          externalId: invoice.paymentRequest,
+          currency: payment.currency,
+          data: Object.assign({}, { invoice }) as unknown as Prisma.InputJsonValue,
+          fee: 0,
+          refunded: false,
+          success: false,
+        },
+      });
+
+      if (!paymentData) {
+        throw new Error();
+      }
+      return paymentData;
+    } catch (error) {
+      console.error(error);
+      throw new Error("Payment could not be created");
+    }
+  }
+  async update(): Promise<Payment> {
+    throw new Error("Method not implemented.");
+  }
+  async refund(): Promise<Payment> {
+    throw new Error("Method not implemented.");
+  }
+
+  async collectCard(
+    payment: Pick<Prisma.PaymentUncheckedCreateInput, "amount" | "currency">,
+    bookingId: number,
+    _bookerEmail: string,
+    paymentOption: PaymentOption
+  ): Promise<Payment> {
+    throw new Error("Method not implemented");
+  }
+  chargeCard(
+    payment: Pick<Prisma.PaymentUncheckedCreateInput, "amount" | "currency">,
+    bookingId: number
+  ): Promise<Payment> {
```

Comment:
We should fix these eslint warnings. I am okay ignoring these warnings directly per line.

Reply:
Also, I am not sure we really need these definitions in there if they are just noop. Maybe we can remove these methods.

Reply:
the method is implemented for stripe, and is part of the interface. The methods could be made optional there, or I can just add ignores on those lines in the Alby implementation

Reply:
I fixed the warnings for now by adding `_` prefix on the arguments